src/uwtools/drivers/driver.py

```diff
@@ -43,19 +45,21 @@ def __init__(
         schema_file: Optional[Path] = None,
         controller: Optional[str] = None,
     ) -> None:
-        self._config_full = config if isinstance(config, YAMLConfig) else YAMLConfig(config=config)
-        self._config_full.dereference(
+        config = config if isinstance(config, YAMLConfig) else YAMLConfig(config=config)
+        config.dereference(
             context={
                 **({STR.cycle: cycle} if cycle else {}),
                 **({STR.leadtime: leadtime} if leadtime is not None else {}),
-                **self._config_full.data,
+                **config.data,
             }
         )
-        self._config = self._config_full
-        for key in key_path or []:
-            self._config = self._config[key]
+        self._config_full, _ = walk_key_path(config.data, key_path or [])
```

Comment:
After the input config is dereferences, walk the key path and point `self._config_full` to the block containing the driver-name and `platform` sub-blocks.

Reply:
I'm adding some code comments about these, and updating the docstrings for the corresponding property methods to be more explicit about what they're returning.